shell/platform/linux/fl_platform_plugin.cc

```diff
@@ -100,7 +118,21 @@ static FlMethodResponse* clipboard_get_data_async(FlPlatformPlugin* self,
   gtk_clipboard_request_text(clipboard, clipboard_text_cb,
                              g_object_ref(method_call));
 
-  // Will response later.
+  // Will respond later.
+  return nullptr;
+}
+
+// Called when Flutter wants to know if the content of the clipboard is able to
+// be pasted, without actually accessing the clipboard content itself.
+static FlMethodResponse* clipboard_has_strings_async(
+    FlPlatformPlugin* self,
+    FlMethodCall* method_call) {
+  GtkClipboard* clipboard =
+      gtk_clipboard_get_default(gdk_display_get_default());
+  gtk_clipboard_request_text(clipboard, clipboard_text_has_strings_cb,
```

Comment:
@robert-ancell Is there a reason to use the `*_request_*` APIs over the `*_wait_*` APIs? Otherwise this could all just be a call to `gtk_clipboard_wait_is_text_available ()`

Reply:
I'm merging this PR, but feel free to reply if anyone thinks this doesn't seem right and I'm happy to take a look at changing this.

Reply:
The `*_wait_*` methods would block and cause no other Flutter method calls etc to be handled. So the implemented method is correct.

Reply:
Ah good to know, thanks for following up @robert-ancell !